src/ApiService/ApiService/Auth/AuthorizationMiddleware.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Net;
+using System.Reflection;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Azure.Functions.Worker.Middleware;
+
+namespace Microsoft.OneFuzz.Service.Auth;
+
+public sealed class AuthorizationMiddleware : IFunctionsWorkerMiddleware {
+    private readonly IEndpointAuthorization _auth;
+    private readonly ILogTracer _log;
+
+    public AuthorizationMiddleware(IEndpointAuthorization auth, ILogTracer log) {
+        _auth = auth;
+        _log = log;
+    }
+
+    public async Async.Task Invoke(FunctionContext context, FunctionExecutionDelegate next) {
+        var attribute = GetAuthorizeAttribute(context);
+        if (attribute is not null) {
+            var req = await context.GetHttpRequestDataAsync() ?? throw new NotSupportedException("no HTTP request data found");
```

Comment:
How does the middleware work with queue/event grid triggers? Do they even go through the auth flow in the first place?

Reply:
Yes, they go through the same pipeline (in that case the `HttpRequestDataAsync` will return null). We don't use the Authorize attribute on any of those functions.